qiime/sdk/visualizer.py

```diff
@@ -219,14 +219,14 @@ def callable_wrapper(*args, **kwargs):
                 parameter_references)
 
             # TODO use user-configured temp dir
-            with tempfile.TemporaryDirectory('qiime2-temp-') as temp_dir:
-                ret_val = self._callable(output_dir=temp_dir, **view_args)
+            with TempPath(dir=True) as temp_dir:
+                ret_val = self._callable(output_dir=str(temp_dir), **view_args)
```

Comment:
👍 for making `temp_dir` a directory path (`str`) when used in a context manager (that will match `tempfile.TemporaryDirectory`). Is that what you guys are wanting input on?